packages/flame/test/sprite_animation_test.dart

```diff
@@ -90,6 +92,27 @@ void main() {
       expect(counter, 1);
     });
 
+    test(
+        'verify call is being made at the first of the frame for multi-frame animation',
+        () {
+      var timePassed = 0.0;
+      const dt = 0.5;
+      var timesCalled = 0;
+      final sprite = MockSprite();
+      final spriteList = [sprite, sprite, sprite];
+      final animation =
+          SpriteAnimation.spriteList(spriteList, stepTime: 1, loop: false);
+      animation.onFrame = (index) {
+        expect(index, timePassed);
```

Comment:
```suggestion
        expect(timePassed, closeTo(index * 1.0, dt/2));
```

Reply:
By doing this, its causing the test to fail:
```
package:test_api                                expect
test\sprite_animation_test.dart 109:9           main.<fn>.<fn>.<fn>
package:flame/src/sprite_animation.dart 352:18  SpriteAnimation.update
test\sprite_animation_test.dart 114:19          main.<fn>.<fn>

Expected: a numeric value within <0.015> of <1.0>
  Actual: <1.0200000000000007>
   Which:  differs by <0.020000000000000684>
   ```

Reply:
This is probably cheating, but it works....
```
expect(timePassed, closeTo(index * 1.0, index + (dt / 2)));
```

Reply:
Hmm, then it's probably `dt` instead of `dt/2`. This parameter is how close the actual value `timePassed` is to the expected value `index*1.0`. Adding `index+` to the tolerance is not a good idea.

Reply:
Makes sense that it would need to be `dt` if we want to accept any `dt` values, since it can go up to 0.99... and then do one more `dt` tick before it passes the 1.0 threshold.

Reply:
Just `dt` works.